src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.SdkArchiveDiff/FindArchiveDiffs.cs

```diff
@@ -0,0 +1,146 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Task = System.Threading.Tasks.Task;
+
+public class FindArchiveDiffs : Microsoft.Build.Utilities.Task, ICancelableTask
+{
+    public class ArchiveItem
+    {
+        public required string Path { get; init; }
+    }
+
+    [Required]
+    public required ITaskItem BaselineArchive { get; init; }
+
+    [Required]
+    public required ITaskItem TestArchive { get; init; }
+
+    [Output]
+    public ITaskItem[] ContentDifferences { get; set; } = [];
+
+    private CancellationTokenSource _cancellationTokenSource = new();
+    private CancellationToken cancellationToken => _cancellationTokenSource.Token;
+    public void Cancel()
+    {
+        _cancellationTokenSource.Cancel();
+    }
+
+    public override bool Execute()
+    {
+        return Task.Run(ExecuteAsync).Result;
+    }
+
+    public async Task<bool> ExecuteAsync()
+    {
+        var baselineTask = Archive.Create(BaselineArchive.ItemSpec);
+        var testTask = Archive.Create(TestArchive.ItemSpec);
+        Task.WaitAll([baselineTask, testTask], cancellationToken);
+        using var baseline = await baselineTask;
+        using var test = await testTask;
+        var baselineFiles = baseline.GetFileNames();
+        var testFiles = test.GetFileNames();
+        ContentDifferences =
+            GetDiffs(baselineFiles, testFiles, PathWithVersions.Equal, PathWithVersions.GetVersionlessPath, cancellationToken)
+            .Select(FromDiff)
+            .ToArray();
+        return true;
+    }
+
+    static ITaskItem FromDiff((string, DifferenceKind) diff)
+    {
+        var item = new TaskItem(diff.Item1);
+        item.SetMetadata("Kind", Enum.GetName(diff.Item2));
+        return item;
+    }
+
+    public enum DifferenceKind
+    {
+        Added,
+        Removed,
+        Unchanged
+    }
+
+    public static List<(string, DifferenceKind DifferenceKind)> GetDiffs(
```

Comment:
Would you put a comment blurb up here indicating what the algorithm is that is determining whether there are diffs?